### ISSUE TITLE
False negative for invalid option for ignore: [] indentation

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#2394]
Ref: https://github.com/stylelint/stylelint/issues/2392

Confirmed that the following (where `"block"` is an invalid keyword option):

```
{
  "rules": {
    "indentation": [2, {"ignore": ["block"]}]
  }
}
```

doesn't throw an invalid option warning. Instead, it seemingly disables the rule.

As a side note, @m-allanson mentioned the new [`jest-validate`](https://github.com/facebook/jest/tree/master/packages/jest-validate) the other day. It might be worth investigating if there's any scope to make use of that within an options validator?

**Comment by OmgImAlexis** [comment]
From what I can see with 600+ errors I wasn't getting anything in the CLI yet with no errors I get the warning. Both times I was running version `^7.9.0` of stylelint installed via npm.

```
yarn test-css v0.19.1
$ node_modules/stylelint/bin/stylelint.js 'static/css/*.css' 

Invalid Option: Invalid value "block" for option "ignore" of rule "indentation"

error Command failed with exit code 2.
```

**Comment by hudochenkov** [comment]
I get an error running stylelint from master branch:

```
$ ./stylelint/bin/stylelint.js st.css

Invalid Option: Invalid value "block" for option "ignore" of rule "indentation"
```

But I have no errors at [demo page](https://stylelint.io/demo/). Maybe it's a problem with a demo and not with stylelint?

Also tried:

```
{
  "rules": {
    "max-empty-lines": "fsdfsdf"
  }
}
```

CLI shows an error, demo — no warnings.

**Comment by m-allanson** [comment]
@hudochenkov The demo should be running stylelint v7.9.0.  

What happens when you test with stylelint v7.9.0 from the CLI? Could the problem have been fixed on master? [git-bisect](https://git-scm.com/docs/git-bisect) is handy for tracking down things like this.

**Comment by hudochenkov** [comment]
@m-allanson checked with 7.9.0 (installed locally, not globally). Got an error.

**Comment by m-allanson** [comment]
@hudochenkov You're right there's a problem with the demo - it does not report invalid configs. I'll create a separate issue there.

**Comment by m-allanson** [comment]
I can't replicate this issue using stylelint v7.9.0 on the CLI. 

I've tried in an empty project, and also by modifying the [system-test 005](https://github.com/stylelint/stylelint/tree/6a260b82ab0567a5a28efe7c18c4aa9b09e4475b/system-tests/005) config to include `"indentation": [2, {"ignore": ["block"]}]`. Then from the `005` dir running `
../../bin/stylelint.js "stylesheet.css" --config config.json > testOutput.txt`. This generates over 600 errors but still includes the line `Invalid Option: Invalid value "block" for option "ignore" of rule "indentation".` at the beginning of the output.

@OmgImAlexis are you able to create an example repo that demonstrates the issue?

**Comment by OmgImAlexis** [comment]
@m-allanson not really as I only get the message when no issues were present and I don't know how many it needed before it stopped showing. At 600+ error we had no message showing.

**Comment by m-allanson** [comment]
Ok, let's see if @jeddy3 can still replicate the problem when back from holiday.

**Comment by jeddy3** [comment]
> Ok, let's see if @jeddy3 can still replicate the problem when back from holiday.

I'm back.

Thanks @m-allanson and @hudochenkov for investigating the issue. As I can't replicate this either, and as we're tracking the demo issue in https://github.com/stylelint/stylelint-demo/issues/27, I'm going to close this.

@OmgImAlexis If you are able to, at some point, create a reproducible test repo feel free to post it below.
